corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -761,38 +761,44 @@ def _record_timing(self, status):
                 duration,
             )
         is_webapps = device_id and device_id.startswith("WebAppsLogin")
-        tags = [
-            'status_code:{}'.format(status),
-            'device_type:{}'.format('webapps' if is_webapps else 'other'),
-        ]
+        tags = {
+            'status_code': status,
+            'device_type': 'webapps' if is_webapps else 'other',
+        }
         maybe_add_domain_tag(self.domain, tags)
         timer_buckets = (1, 5, 20, 60, 120, 300, 600)
         for timer in timing.to_list(exclude_root=True):
+            segment = None
+            extra_tags = {}
             if timer.name in RESTORE_SEGMENTS:
                 segment = RESTORE_SEGMENTS[timer.name]
-                bucket = bucket_value(timer.duration, timer_buckets, 's')
-                datadog_counter(
-                    'commcare.restores.{}'.format(segment),
-                    tags=tags + ['duration:%s' % bucket],
-                )
             elif timer.name.startswith('fixture:'):
-                bucket = bucket_value(timer.duration, timer_buckets, 's')
-                datadog_counter(
-                    'commcare.restores.fixture',
-                    tags=tags + [
-                        'duration:%s' % bucket,
-                        timer.name,
-                    ],
+                segment = 'fixture'
+                extra_tags = {'fixture': timer.name.split(':')[1]}
+
+            if segment:
+                metrics_histogram(
+                    'commcare.restores.{}.duration.seconds'.format(segment), timer.duration,
+                    bucket_tag='duration', buckets=timer_buckets, bucket_unit='s',
+                    tags={**tags, **extra_tags}
+                )
+                metrics_counter(
+                    'commcare.restores.{}'.format(segment),
+                    tags={**tags, **extra_tags},
```

Comment:
`commcare.restores.*` split into a counter and a 'bucket histogram'

Reply:
yes it will. I already have a script to update the dashboards which I've been using up till now. For the most part it's just find and replace.